Update backup-azure-backup-faq.md

MattiasAng opened the pull request:
Added documentation due to new feature that allows migration from backup vault to recovery services vault.

azurecla commented:
Hi __@raymondlaghaeian__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

MattiasAng committed:
Update backup-azure-backup-faq.md

Added documentation due to new feature that allows migration from backup vault to recovery services vault.
